Tools/LyTestTools/ly_test_tools/o3de/material_editor_test.py

```diff
@@ -0,0 +1,825 @@
+    as well as serially batched with other tests in each MaterialEditor instance.
+    Minimizes total test run duration.
+
+    Does not support per test setup/teardown to avoid creating race conditions
+    """
+    # Specifies if the test can be batched in the same editor
+    is_batchable = True
+    # Specifies if the test can be run in multiple editors in parallel
+    is_parallelizable = True
+
+
+class MaterialEditorParallelTest(MaterialEditorSharedTest):
+    """
+    Test that will be run in parallel with tests in different MaterialEditor instances,
+    though not serially batched with other tests in each MaterialEditor instance.
+    Reduces total test run duration, while limiting side-effects between tests.
+
+    Does not support per test setup/teardown to avoid creating race conditions
+    """
+    is_batchable = False
+    is_parallelizable = True
+
+
+class MaterialEditorBatchedTest(MaterialEditorSharedTest):
+    """
+    Test that will be batched along with the other batched tests in the same MaterialEditor instance,
+        though not executed in parallel with other MaterialEditor instances.
+    Reduces repeated overhead from starting the MaterialEditor.
+
+    Does not support per test setup/teardown to avoid creating race conditions
+    """
+    is_batchable = True
+    is_parallelizable = False
+
+
+@pytest.mark.parametrize("crash_log_watchdog", [("raise_on_crash", False)])
+class MaterialEditorTestSuite(AbstractTestSuite):
+    # Extra cmdline arguments to supply for every editor instance for this test suite
+    global_extra_cmdline_args = ["-BatchMode", "-autotest_mode"]
+    # Tests usually run with no renderer, however some tests require a renderer
+    use_null_renderer = True
+    # Maximum time for a single editor to stay open on a shared test
+    timeout_editor_shared_test = 300
+    # Maximum time (seconds) for waiting for a crash file, in seconds
+    timeout_crash_log = 20
+    # Return code for test failure
+    test_fail_return_code = 0xF
+    # Test class to use for single test collection
+    single_test_class = MaterialEditorSingleTest
+    # Test class to use for shared test collection
+    shared_test_class = MaterialEditorSharedTest
+
+    @classmethod
+    def get_instance_log_str(cls):
+        # type () -> str
+        """
+        Checks if a given material_editor_log attribute exists and returns it.
+        :return: Either the object attribute matching "material_editor_log" (typically log output) or None.
+        """
+        log = getattr(cls, "material_editor_log", None)
+        if log:
+            return log
+        else:
+            return "-- No material_editor_log found --"
+
+    class MaterialEditorTestClass(pytest.Class):
+        """
+        Pytest collector which programmatically adds test functions based on data in the MaterialEditorTestSuite class
+        """
+
+        def collect(self):
```

Comment:
this should be possible to make generic in the abstract base class

Reply:
as discussed on Discord, we'll go over this together when you have time to